src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+package mate.academy.dao.impl;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String FIND_ALL_QUERY = "SELECT * FROM book;";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM book WHERE id = ?;";
+    private static final String CREATE_QUERY = "INSERT INTO book (title, price) VALUES (?, ?);";
+    private static final String UPDATE_QUERY = "UPDATE book SET title = ?, price = ? WHERE id = ?;";
```

Comment:
The table name should be `books` instead of `book`. Remember to use the correct table name as per SQL conventions.